From 5566bd132ffa83cbd3360c08dccef88fb17a9edf Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Thu, 22 Jul 2021 15:47:07 +0200
Subject: [PATCH] Disable read README

---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index 456a5cb4..f85d0e47 100644
--- a/setup.py
+++ b/setup.py
@@ -82,7 +82,7 @@ def get_package_xml_version():
 
 
 with open('README.md', 'r') as f:
-    long_description = f.read()
+    long_description = ""
 
 setup(
     name='manifpy',
