From 3a07d08e996bc822234cd536ebc279cabd37c4d5 Mon Sep 17 00:00:00 2001
From: Silvio Traversaro <silvio.traversaro@iit.it>
Date: Tue, 15 Jun 2021 13:46:23 +0200
Subject: [PATCH] Bump CMake version to 0.0.4

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index e9ed4a38..a50fd0cf 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,7 +1,7 @@
 cmake_minimum_required(VERSION 3.5.1)
 cmake_policy(SET CMP0048 NEW)
 
-project(manif VERSION 0.0.3 LANGUAGES CXX)
+project(manif VERSION 0.0.4 LANGUAGES CXX)
 
 # Get the project name in capital letters
 # for the lib namespace
 
